getting-started/quickstart-guides/angular.md

```diff
@@ -0,0 +1,834 @@
+
+The response object would look similar to this (values may vary):
+
+```js
+{
+    "fileId": "63a2e985e809dd54b027a563",
+    "name": "test_RiCBw0ouh.jpg",
+    "size": 36919,
+    "versionInfo": {
+        "id": "63a2e985e809dd54b027a563",
+        "name": "Version 1"
+    },
+    "filePath": "/test_RiCBw0ouh.jpg",
+    "url": "https://ik.imagekit.io/yzyzyz/test_RiCBw0ouh.jpg",
+    "fileType": "image",
+    "height": 500,
+    "width": 500,
+    "thumbnailUrl": "https://ik.imagekit.io/yzyzyz/tr:n-ik_ml_thumbnail/test_RiCBw0ouh.jpg",
+    "AITags": null
+}
+```
+
+After a successful upload, you should see the newly uploaded image in the [Media Library](https://imagekit.io/dashboard#media-library) section of your ImageKit dashboard.
+
+If you don't see the image, check if there are any errors in the browser console log. Then verify whether the API [private key](../../api-reference/api-introduction/api-keys.md#private-key) has been configured correctly in the server app and if the server app is running.
+
+**Fetching uploaded file**
+
+Fetch uploaded image and show in UI using `ik-image` with the `filePath` returned in the upload response.
+
+```jsx
+<ik-image path="/test_RiCBw0ouh.png" ></ik-image>
+```
+
+The app should display your uploaded image correctly!
+
+## **Advanced file upload**
+
+A more detailed example of how to use the file upload component (and an explanation of each advanced feature) is presented below:
+
+{% code title="src/app/app.component.ts" %}
+```js
+import { Component } from '@angular/core';
+import { Transformation } from 'imagekit-javascript/dist/src/interfaces/Transformation';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+    validateFileFunction(res: any) {
+    console.log('validating')
+    if(res.size < 1000000){ // Less than 1mb
+      return true;
+    }
+    return false;
+  }
+
+  onUploadStartFunction(res: any) {
+    console.log('onUploadStart')
+  }
+
+  onUploadProgressFunction(res: any) {
+    console.log('progressing')
+  }
+
+  handleUploadSuccess(res: any) {
+    console.log('File upload success with response: ', res);
+    console.log(res.$ResponseMetadata.statusCode); // 200
+    console.log(res.$ResponseMetadata.headers); // headers
+    this.uploadedImageSource = res.url;
+  }
+
+  handleUploadError(err: any) {
+    console.log('There was an error in upload: ', err);
+    this.uploadErrorMessage = 'File upload failed.';
+  }
+}
+```
+{% endcode %}
+
+{% code title="src/app/app.component.html" %}
+{% endcode %}
+
+### **Custom Upload Button**
+We have created a `ref` to the `input` used inside the `ik-upload` component called `inputRefTest`. The `ik-upload` component can be given styling via `className` or `style` (`style={{display: 'none'}}`) to hide the default file selector. Then we can use the custom upload button as described above.
```

Comment:
This seems to be a blind copy paste from React and does not make sense here at all. Where is the code above or any mention of `inputRefTest`?

Reply:
Removed

Reply:
Updated with buttonRef which I have sent a video to you of before

Reply:
Updated wording